### PR TITLE
Change to the NetBSD archive URL rather than the CDN

### PR DESCRIPTION
**Pull request by tgross35**
The CDN has been down for a few hours. Switch to an alternative for the time being so we can unblock CI.

It appears that the CDN is quite a bit faster, so we will likely want to switch back when available.

Context: https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/dist-x86_64-netbsd.20is.20borked

try-job: dist-x86_64-netbsd

### CODE REVIEW

File: src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh [hunk 1]
```diff
@@ -27,7 +27,10 @@ mkdir -p /x-tools/x86_64-unknown-netbsd/sysroot
 
 # URL=https://ci-mirrors.rust-lang.org/rustc
 
-SOURCE_URL=https://cdn.netbsd.org/pub/NetBSD/NetBSD-9.0/source/sets
+# FIXME: the archive URL is being used temporarily while the CDN is down.
+# Change back to the original URL once it is back up.
+# SOURCE_URL=https://cdn.netbsd.org/pub/NetBSD/NetBSD-9.0/source/sets
+SOURCE_URL=http://archive.netbsd.org/pub/NetBSD-archive/NetBSD-9.0/source/sets/src.tgz
```
Comment (Urgau) [inline src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh:33]: shouldn't this be?
```suggestion
SOURCE_URL=http://archive.netbsd.org/pub/NetBSD-archive/NetBSD-9.0/source/sets
```

Comment (Urgau) [inline src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh:33]: what about line 39? it's also using still using the cdn

File: src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh [hunk 2]
```diff
@@ -27,7 +27,10 @@ mkdir -p /x-tools/x86_64-unknown-netbsd/sysroot
 
 # URL=https://ci-mirrors.rust-lang.org/rustc
 
-SOURCE_URL=https://cdn.netbsd.org/pub/NetBSD/NetBSD-9.0/source/sets
+# FIXME: the archive URL is being used temporarily while the CDN is down.
+# Change back to the original URL once it is back up.
```
Comment (Nilstrieb) [inline src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh:31]: we should not do that, we should instead mirror it ourselves and use that URL.

File: src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh (hunk 1 above)
Comment (tgross35) [reply src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh]: Yes that's correct, fixed

Comment (tgross35) [reply src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh]: I added this too

File: src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh (hunk 2 above)
Comment (tgross35) [reply src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh]: Right, I updated the fixme to say that

**Comment by tgross35** [comment]
Just realized the archive URL uses http and doesn't seem to have https, we may not want this then :/

I'll give it another half hour to see if the CDN comes back. If not, I'll add a SHA sum check. Nothing to calculate the correct SHA from except what's currently available at the archive URL, unfortunately... Open to suggestions

**Comment by tgross35** [comment]
CDN is still down. I added a check against the sha from @notriddle's link.

Could somebody `@bors try` if this looks correct?

**Comment by tgross35** [comment]
Why no dist-x86_64-netbsd in the try, it's in the commit message 🤔

**Comment by tgross35** [comment]
Does `<!-- homu-ignore:start --> .. <!-- homu-ignore:end -->` make bors ignore try-job directives too?  (my try-job was wrapped in those). Thought that was just what gets stripped from the commit message. Removed those comments in case that's it.

**Comment by Kobzol** [comment]
I think that the problem is that the stuff inside the comments is indeed ignored by homu, and it then doesn't add it to the commit message. You can see that it is not here: https://github.com/rust-lang/rust/commit/95b682c9c37829620cf9d29d5d4fbfc1e793b0a4.

CI reads the commit message, which is generated from the PR description by `homu`, but here it was ignored.

**Comment by tgross35** [comment]
_Oh_ that makes more sense, thanks. I removed the comments, mind restarting the try?

**Comment by tgross35** [comment]
Forgot to rename a function, could use another try please

**Comment by Nilstrieb** [comment]
try harder
@bors try

File: src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh [hunk 3]
```diff
@@ -20,24 +20,54 @@ exit 1
...
+
+# FIXME: the archive URL is being used temporarily while the CDN is down.
+# We should serve this from our own CDN
+# BINARY_URL=https://cdn.netbsd.org/pub/NetBSD/NetBSD-9.0/amd64/binary/sets
+BINARY_URL=http://archive.netbsd.org/pub/NetBSD-archive/NetBSD-9.0/amd64/binary/sets
+download base.tar.xz "$BINARY_URL/base.tar.xz" "$BASE_SHA" \
   tar xJf - -C /x-tools/x86_64-unknown-netbsd/sysroot ./usr/include ./usr/lib ./lib
-curl $BINARY_URL/comp.tar.xz | \
-  tar xJf - -C /x-tools/x86_64-unknown-netbsd/sysroot ./usr/include ./usr/lib
+download comp.tar.xz "$BINARY_URL/comp.tar.xz" "$COMP_SHA" \
```
Comment (DianQK) [inline src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh:69]:
```suggestion
download comp.tar.xz "$BINARY_URL/comp.tar.xz" "$COMP_SHA"
```
Here?

Comment (tgross35) [reply src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh:69]: Thanks, just got the same :)

Comment (DianQK) [reply src/ci/docker/host-x86_64/dist-x86_64-netbsd/build-netbsd-toolchain.sh:69]: Sorry, I didn't notice that the download function have included this command. :3

**Comment by tgross35** [comment]
Wound up just testing with a CI job, which downloaded everything okay and seemed to be continuing fine till dropping the CI commit cancaled it, https://github.com/rust-lang/rust/actions/runs/9757991845/job/26931626628?pr=127232.

So this should be ready to go if it is an acceptable short term solution. Still seems weird that netbsd's mirror is still seemingly borked for 9.0 and 9.1 in particular.

**Comment by Kobzol** [comment]
We should upload these files to our mirrors, but for now this is good to unblock CI. The initial Docker build has indeed completed fine in the try build, but I want to see first if the whole try build completes. Once it does, I'll merge it.

**Comment by Kobzol** [comment]
@bors r+ p=101

(should unblock CI, hopefully)

**Comment by tgross35** [comment]
How could we be hitting the rate limit when there hasn't been any CI for 8 hours?

**Comment by Kobzol** [comment]
Good idea. Normally, the builds are cached, so it doesn't even download the Docker images, but they are busted quite often, and obviously PR CI works, which also downloads them. No idea what's going on, can't debug at the moment. Will try to take a look later.

**Comment by Kobzol** [comment]
@bors retry

(Just trying if it was perhaps trainsient Docker issue..)

**Comment by Kobzol** [comment]
These spurious failures are really starting to compound...

@bors retry

**Comment by cuviper** [comment]
I'm unilaterally backporting this to unblock #127377.

@rustbot label +beta-accepted

**Comment by tgross35** [comment]
I'm surprised that it is still down.

@he32 (since you're listed as the NetBSD maintainer) any clue if anyone from NetBSD are aware of this? Summary: the CDN for 9.0 and 9.1 have been 404ing for a few days (e.g. https://cdn.netbsd.org/pub/NetBSD/NetBSD-9.0/source/sets/). Other versions correctly redirected to the archive URL for a while while the CDN was down, and now seem to be back up at the CDN (e.g. https://cdn.netbsd.org/pub/NetBSD/NetBSD-9.2/source/sets/) - but 9.0 and 9.1 CDN links are just dead.

This isn't currently a problem for Rust since we switched to the archive URL (this PR), but presumably we aren't the only ones to experience some fallout.

**Pull request merged**